### ISSUE TITLE
Standalone Command Line Auto Filling

### ISSUE THREAD

**Issue by the-monkey-john** [inkstitch/inkstitch#3187]
Hi, I'd like to use just command line with the inkstitch extension binary (inkscape shouldn't be necessary) to turn an SVG into a PES or whatever output format I choose with the abililty to set various options e.g. minimum stitch length, distance between stitch lines, etc. Ideally, I could also choose different types of fills e.g. circular. Is this currently possible? https://github.com/inkstitch/inkstitch/issues/1913 seems to suggest this is possible but there isn't very good documentation.

**Comment by lexelby** [comment]
Yup, this is possible, although it may not be quite how you're thinking of it.  Here's the documentation:

https://inkstitch.org/docs/command-line/

You would need to prepare the SVG by setting XML attributes in the same way that Params does. You could try running Params manually and looking at the SVG to see how various settings translate to XML attributes.

**Comment by the-monkey-john** [comment]
Ah, thanks! Do you know if there is a command I could use to set the XML attributes for the SVG? If not, a code pointer to where this is done programmatically would be appreciated so I can replicate it myself in python.

**Comment by lexelby** [comment]
It's done in lib/elements/element.py, in EmbroideryElement.set_param(), I believe.  The important bit is to make sure you get the XML namespace right.  You should be able to do this using the builtin python lxml module, or you could use inkex to load and modify the SVG instead. inkex is more geared toward SVG files rather than generic XML like lxml.

**Comment by kaalleen** [comment]
The way Ink/Stitch handles namespaces is pretty outdated by now. Inkex can handle them much better these days. So it'd be better and easier to stick to the inkex documentation rather than the Ink/Stitch source code.
